cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java

```diff
@@ -389,7 +389,7 @@ private DescribeVpcsRequest getDescribeVpcsRequestWIthFilters(Map<String, String
                             subnetName.orElse(subnet.getSubnetId()),
                             subnet.getAvailabilityZone(),
                             subnet.getCidrBlock(),
-                            !subnet.isMapPublicIpOnLaunch(),
+                            !hasInternetGateway,
```

Comment:
We are passing hasInternetGateway's value 2 times here.

Reply:
thats ok, as private subnet flag is driven by it

Reply:
Ok, I see the value in that and addressed it with a new ctor. However, it would be great to refactor CloudSubnet as it has too many ctors.